sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v2/query.sql

```diff
@@ -0,0 +1,404 @@
+CREATE TEMP FUNCTION normalize_browser(browser STRING) AS (
+  CASE
+    WHEN `moz-fx-data-shared-prod.udf.ga_is_mozilla_browser`(browser)
+      THEN 'Firefox'
+    WHEN browser IN ('Internet Explorer')
+      THEN 'MSIE'
+    WHEN browser IN ('Edge')
+      THEN 'Edge'
+    WHEN browser IN ('Chrome')
+      THEN 'Chrome'
+    WHEN browser IN ('Safari')
+      THEN 'Safari'
+    WHEN browser IN ('(not set)')
+      THEN NULL
+    WHEN browser IS NULL
+      THEN NULL
+    ELSE 'Other'
+  END
+);
+
+-- Unnest all the hits for all  sessions, one row per hit.
+-- Different hit.type values (EVENT | PAGE)
+-- and hit.eventInfo.eventAction('Firefox Download'| 'Stub Session ID') are
+-- extracted in the following CTEs
+-- Those extracted fields are joined to the GA session level data, one session per day.
+WITH all_hits AS (
+  SELECT
+    clientId AS client_id,
+    visitId AS visit_id,
+    hit.appInfo.landingScreenName AS landing_page,
+    hit.page.pagePath AS pagepath,
+    CASE
```

Comment:
Since this appears to be a true or false expression. Would it be easier to read if expressed using `if()`?

Reply:
+1